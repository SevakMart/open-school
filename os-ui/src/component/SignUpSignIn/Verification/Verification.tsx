import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { useState } from 'react';
import { openModalWithSuccessMessage } from '../../../redux/Slices/PortalOpenStatus';
import { Types } from '../../../types/types';
import Button from '../../Button/Button';
import Header from '../Header/Header';
import { validateEmail } from '../../../helpers/EmailValidate';
import authService from '../../../services/authService';
import { addEmail } from '../../../redux/Slices/ForgotPasswordEmailSlice';

type ForgotVerificationProps = {
	forgotVerficationEmail?: string
}
/* eslint-disable max-len */
const Verification = (props: ForgotVerificationProps) => {
  const { forgotVerficationEmail = '' } = props;
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const [emailError, setEmailError] = useState('');

  const handleResend = () => {
    const emailError = validateEmail(forgotVerficationEmail);
    if (!emailError) {
      authService.sendForgotVerificationRequest({ email: forgotVerficationEmail })
        .then((response) => {
          if (response.status === 200) {
            const successMessage = t('Verification Email Sent!');
            dispatch(openModalWithSuccessMessage({
              buttonType: Types.Button.SUCCESS_MESSAGE,
              withSuccessMessage: successMessage,
              isForgotVerificationMessage: true,
            }));
            dispatch(addEmail(forgotVerficationEmail));
            setEmailError('');
          }
        });
    } else setEmailError(emailError);
  };

  return (
    <>
      <Header
        mainTitle={t('string.onAccountVerification.title')}
        shouldRemoveIconContent
        isForgotPasswordContent={false}
        isVerificationContent
      />
      {emailError && <span style={{ color: 'red' }}>{emailError}</span>}
      <div style={{ margin: '0 80px' }}>
        <Button.FormButton className={['formButton']} onClick={handleResend}>
          {t('button.resend')}
        </Button.FormButton>
      </div>
    </>
  );
};
export default Verification;
