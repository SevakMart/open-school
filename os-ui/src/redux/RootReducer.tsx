import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
