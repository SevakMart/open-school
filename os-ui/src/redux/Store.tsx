import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
