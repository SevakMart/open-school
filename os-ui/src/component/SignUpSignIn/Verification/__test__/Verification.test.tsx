import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../redux/Store';
import Verification from '../Verification';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test case for Verification component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Verification />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
