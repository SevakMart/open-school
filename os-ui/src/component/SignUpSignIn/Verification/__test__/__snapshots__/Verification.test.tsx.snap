// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test case for Verification component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="closeIcon"
  >
    âœ–
  </div>
  <div
    class="headerContent"
  >
    <h2>
      string.onAccountVerification.title
    </h2>
    <p>
      messages.verificationPageHint
    </p>
  </div>
  <div
    style="margin: 0px 80px;"
  >
    <button
      class="formButton"
      type="submit"
    >
      button.resend
    </button>
  </div>
</DocumentFragment>
`;
